tm2/pkg/sdk/auth/ante.go

```diff
@@ -379,43 +379,54 @@ func DeductFees(bank BankKeeperI, ctx sdk.Context, acc std.Account, fees std.Coi
 // consensus.
 func EnsureSufficientMempoolFees(ctx sdk.Context, fee std.Fee) sdk.Result {
 	minGasPrices := ctx.MinGasPrices()
+	blockHeader := ctx.BlockHeader().(*bft.Header)
+
+	blockGasPrice := std.GasPrice{
+		Gas: blockHeader.GasPriceGas,
+		Price: std.Coin{
+			Amount: blockHeader.GasPriceAmount,
+			Denom:  blockHeader.GasPriceDenom,
+		},
+	}
+
+	feeGasPrice := std.GasPrice{
+		Gas: fee.GasWanted,
+		Price: std.Coin{
+			Amount: fee.GasFee.Amount,
+			Denom:  fee.GasFee.Denom,
+		},
+	}
+	// check the block gas price
+	if blockGasPrice.Price.IsValid() && !blockGasPrice.Price.IsZero() {
+		if feeGasPrice.IsGTE(blockGasPrice) == false {
+			return abciResult(std.ErrInsufficientFee(
+				fmt.Sprintf(
+					"insufficient fees; got: %q required: %q as block gas price", fee.GasFee, blockGasPrice,
+				),
+			))
+		}
+	}
+	// check min gas price set by the node.
 	if len(minGasPrices) == 0 {
 		// no minimum gas price (not recommended)
 		// TODO: allow for selective filtering of 0 fee txs.
 		return sdk.Result{}
 	} else {
-		fgw := big.NewInt(fee.GasWanted)
-		fga := big.NewInt(fee.GasFee.Amount)
-		fgd := fee.GasFee.Denom
-
 		for _, gp := range minGasPrices {
-			gpg := big.NewInt(gp.Gas)
-			gpa := big.NewInt(gp.Price.Amount)
-			gpd := gp.Price.Denom
-
-			if fgd == gpd {
-				prod1 := big.NewInt(0).Mul(fga, gpg) // fee amount * price gas
-				prod2 := big.NewInt(0).Mul(fgw, gpa) // fee gas * price amount
-				// This is equivalent to checking
-				// That the Fee / GasWanted ratio is greater than or equal to the minimum GasPrice per gas.
-				// This approach helps us avoid dealing with configurations where the value of
-				// the minimum gas price is set to 0.00001ugnot/gas.
-				if prod1.Cmp(prod2) >= 0 {
-					return sdk.Result{}
-				} else {
-					return abciResult(std.ErrInsufficientFee(
-						fmt.Sprintf(
-							"insufficient fees; got: %q required: %q", fee.GasFee, gp,
-						),
-					))
-				}
+			if feeGasPrice.IsGTE(gp) {
+				return sdk.Result{}
+			} else if feeGasPrice.Price.Denom == gp.Price.Denom {
```

Comment:
Thank you for the recommendation. I will do bit more to make it as ` if feeGasPrice.IsLT(gp) && feeGasPrice.Price.Denom == gp.Price.Denom {`

Reply:
Thank you for the recommendation. If this were just a single gas price comparison, I would agree. However, the gas price is compared with a list of` minPrices `across multiple denoms. This emphasizes the logic: if the gas price meets one of the settings, we allow it. That is why I wrote the condition separately as a positive case, instead of making it implicit as the outcome of a negative case in a for loop.